core/src/main/java/org/elasticsearch/search/suggest/Suggest.java

```diff
@@ -399,35 +400,17 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                         "Cannot parse suggestion response without type information. Set [" + RestSearchAction.TYPED_KEYS_PARAM
                                 + "] parameter on the request to ensure the type information is added to the response output");
             }
-            Suggestion suggestion = null;
-            Function<XContentParser, Entry> entryParser = null;
-            // the "size" parameter and the SortBy for TermSuggestion cannot be parsed from the response, use default values
-            // TODO investigate if we can use NamedXContentRegistry instead of this switch
-            switch (type) {
-                case Suggestion.NAME:
```

Comment:
I do not understand what this is for. you can have a custom suggester but it has to be called "suggestion"? This makes no sense to me. If it is possible to register custom suggesters (I hear it may not be), RestHighLevelClient has to be subclassed and the custom parser has to be added to the registry, that's it. Note that these parsers will be added to the registry in a followup PR, as that change will go only to master.